### PR TITLE
feat(utxo-lib): implement mweb parsing for ltc

### PR DESCRIPTION
**Pull request by davidkaplanbitgo**
Litecooin transactions are able to deserialize with a MWEB advanced transaction flag. We always reserialize the transaction without the MWEB flag

BG-74919

### CODE REVIEW

File: modules/utxo-lib/src/bitgo/litecoin/LitecoinTransaction.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    const bufferReader = new BufferReader(buffer);
+    const tx = new LitecoinTransaction<TNumber>(network);
+    tx.version = bufferReader.readInt32();
+    const marker = bufferReader.readUInt8();
+    const flag = bufferReader.readUInt8();
+
+    let hasWitnesses = false;
+
+    if (isLitecoin(network)) {
```
Comment (brandonblack) [inline modules/utxo-lib/src/bitgo/litecoin/LitecoinTransaction.ts:57]: Why do we need this check when we're in `LitecoinTransction`?

File: modules/utxo-lib/test/bitgo/litecoin/mwebBehavior.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  createTransactionFromHex,
+  LitecoinPsbt,
+  LitecoinTransaction,
+  NonWitnessWalletUnspent,
+} from '../../../src/bitgo';
+import { getDefaultWalletKeys, mockWalletUnspent } from '../../../src/testutil';
+
+const network = networks.litecoin;
+
+async function getFixture<T>(name: string): Promise<T> {
```
Comment (mmcshinsky-bitgo) [inline modules/utxo-lib/test/bitgo/litecoin/mwebBehavior.ts:21]: Has this been tested to make sure it works properly in a web environment like the `web-demo` module?

File: modules/utxo-lib/test/bitgo/litecoin/fixtures/ltc-mweb-transaction.json [hunk 3]
```diff
@@ -0,0 +1 @@
+"02000000000801c9acff2c66910bba2faca1b7c6889f5f69e2ac04ff42452ccbb05e84d1dd55f90000000000ffffffff0287cec866e9000000225820bce2cc319d7393fdb728489d1c3060b129b70ef20304af59d88d0def48669e63100905000000000017a9143c006dfa10642aa0e7f7171c2a3afb78f7529e96870000000000"
```
Comment (davidkaplanbitgo) [inline modules/utxo-lib/test/bitgo/litecoin/fixtures/ltc-mweb-transaction.json:1]: Find MWEB ltc transactions by looking [on the explorer](https://www.mwebexplorer.com/blocks) by finding blocks that have MWEB transactions

File: modules/utxo-lib/src/bitgo/litecoin/LitecoinTransaction.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    if (
+      marker === LitecoinTransaction.ADVANCED_TRANSACTION_MARKER &&
+      flag === LitecoinTransaction.MWEB_PEGOUT_TX_FLAG
+    ) {
+      // Litecoin has an MWEB advanced transaction marker. Slice out the marker and 5th to last byte  and read like a normal transaction
+      const bufferWriter = new BufferWriter(Buffer.allocUnsafe(buffer.length - 3));
+      bufferWriter.writeUInt32(txVersion);
+      bufferWriter.writeSlice(buffer.slice(6, buffer.length - 5));
+      bufferWriter.writeSlice(buffer.slice(buffer.length - 4, buffer.length));
+      return super.fromBuffer(bufferWriter.buffer, noStrict, amountType, network);
```
Comment (OttoAllmendinger) [inline modules/utxo-lib/src/bitgo/litecoin/LitecoinTransaction.ts:59]: :+1:

**Review by OttoAllmendinger — APPROVED** [review]

File: modules/utxo-lib/src/bitgo/wallet/Unspent.ts [hunk 5]
```diff
@@ -163,6 +164,10 @@ export function addWalletUnspentToPsbt(
     if (!isUnspentWithPrevTx(u)) {
       throw new Error('Error, require previous tx to add to PSBT');
     }
+    // Force the litecoin transaction to have no MWEB advanced transaction flag
+    if (getMainnet(psbt.network) === networks.litecoin) {
+      u.prevTx = createTransactionFromBuffer(u.prevTx, psbt.network, { amountType: 'bigint' }).toBuffer();
```
Comment (brandonblack) [inline modules/utxo-lib/src/bitgo/wallet/Unspent.ts:169]: gross, but makes sense.

**Review by brandonblack — APPROVED** [review]

**Pull request merged**